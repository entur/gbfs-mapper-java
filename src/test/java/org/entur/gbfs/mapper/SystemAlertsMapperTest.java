package org.entur.gbfs.mapper;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.io.IOException;
import java.net.URL;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@ExtendWith({SnapshotExtension.class})
class SystemAlertsMapperTest {
    private static final ObjectMapper objectMapper = new ObjectMapper();
    private Expect expect;

    @SnapshotName("gbfs_v2_3_to_v3_0_system_alerts_file_snapshot")
    @Test
    void testMapSystemAlertsFile() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v2_3/system_alerts.json");
        org.mobilitydata.gbfs.v2_3.system_alerts.GBFSSystemAlerts testSubject = objectMapper.readValue(resource, org.mobilitydata.gbfs.v2_3.system_alerts.GBFSSystemAlerts.class);
        org.mobilitydata.gbfs.v3_0.system_alerts.GBFSSystemAlerts mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }

    @SnapshotName("gbfs_v3_0_to_v2_3_system_alerts_file_snapshot")
    @Test
    void testMapSystemAlertsFileInverse() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v3_0/system_alerts.json");
        org.mobilitydata.gbfs.v3_0.system_alerts.GBFSSystemAlerts testSubject = objectMapper.readValue(resource, org.mobilitydata.gbfs.v3_0.system_alerts.GBFSSystemAlerts.class);
        org.mobilitydata.gbfs.v2_3.system_alerts.GBFSSystemAlerts mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }
}
