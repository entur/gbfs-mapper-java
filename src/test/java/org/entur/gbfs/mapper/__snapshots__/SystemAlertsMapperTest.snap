gbfs_v2_3_to_v3_0_system_alerts_file_snapshot=[
  {
    "last_updated" : "1970-01-19T13:41:59.393+00:00",
    "ttl" : 60,
    "version" : "3.0-RC2",
    "data" : {
      "alerts" : [
        {
          "alert_id" : "21",
          "type" : "station_closure",
          "times" : [
            {
              "start" : "1970-01-19T13:40:48.000+00:00",
              "end" : "1970-01-19T13:44:34.800+00:00"
            }
          ],
          "station_ids" : [
            "123",
            "456",
            "789"
          ],
          "url" : [
            {
              "text" : "https://example.com/more-info",
              "language" : "en"
            }
          ],
          "summary" : [
            {
              "text" : "Disruption of Service",
              "language" : "en"
            }
          ],
          "description" : [
            {
              "text" : "The three stations on Broadway will be out of service from 12:00am Nov 3 to 3:00pm Nov 6th to accommodate road work",
              "language" : "en"
            }
          ],
          "last_updated" : "1970-01-19T13:36:38.100+00:00"
        }
      ]
    }
  }
]


gbfs_v3_0_to_v2_3_system_alerts_file_snapshot=[
  {
    "last_updated" : 1604519393,
    "ttl" : 60,
    "version" : "2.3",
    "data" : {
      "alerts" : [
        {
          "alert_id" : "21",
          "type" : "station_closure",
          "times" : [
            {
              "start" : 1604448000,
              "end" : 1604674800
            }
          ],
          "station_ids" : [
            "123",
            "456",
            "789"
          ],
          "url" : "https://example.com/more-info",
          "summary" : "Disruption of Service",
          "description" : "The three stations on Broadway will be out of service from 12:00am Nov 3 to 3:00pm Nov 6th to accommodate road work",
          "last_updated" : 1.6041981E9
        }
      ]
    }
  }
]